web/client/actions/users.js

```diff
@@ -0,0 +1,250 @@
+/**
+ * Copyright 2015, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const USERMANAGER_GETUSERS = 'USERMANAGER_GETUSERS';
+const USERMANAGER_EDIT_USER = 'USERMANAGER_EDIT_USER';
+const USERMANAGER_EDIT_USER_DATA = 'USERMANAGER_EDIT_USER_DATA';
+const USERMANAGER_UPDATE_USER = 'USERMANAGER_UPDATE_USER';
+const USERMANAGER_DELETE_USER = 'USERMANAGER_DELETE_USER';
+const USERMANAGER_GETGROUPS = 'USERMANAGER_GETGROUPS';
+const API = require('../api/GeoStoreDAO');
+const {get, assign} = require('lodash');
+
+function getUsers(searchText, options) {
+    let params = options && options.params;
+    let start;
+    let limit;
+    if (params) {
+        start = params.start;
+        limit = params.limit;
+    }
+    return (dispatch, getState) => {
+        let text = searchText;
+        let state = getState && getState();
+        if (state) {
+            let oldText = get(state, "users.searchText");
+            text = searchText || oldText || "*";
+            start = ( (start !== null && start !== undefined) ? start : (get(state, "users.start") || 0));
+            limit = limit || get(state, "users.limit") || 12;
+        }
+        dispatch({
+            type: USERMANAGER_GETUSERS,
+            status: "loading",
+            searchText: text,
+            start,
+            limit
+        });
+
+        return API.getUsers(text, {...options, params: {start, limit}}).then((response) => {
+            let users;
+            // this because _.get returns an array with an undefined element isntead of null
+            if (!response || !response.ExtUserList || !response.ExtUserList.User) {
+                users = [];
+            } else {
+                users = get(response, "ExtUserList.User");
+            }
+
+            let totalCount = get(response, "ExtUserList.UserCount");
+            users = Array.isArray(users) ? users : [users];
+            users = users.map((user) => {
+                let groups = get(user, "groups.group");
+                groups = Array.isArray(groups) ? groups : [groups];
+                return assign({}, user, {
+                    groups
+                });
+            });
+            dispatch({
```

Comment:
Didn't find any convention that forces to create module internal function that create the final object from the action creator. I thought to improve readability in this way (avoid to jump here and there for simple objects definition, avoid passing variables that can be inverted and so on...) I fixed this, let's check if it is a good convention and lets write it if there is a reason to make it mandatory.